### PR TITLE
Refactor parsers formaters

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

This PR extracts all the parsers and formatters to individual packages.

We need to decide if we add the parsers and formatters as direct dependencies of `sonarwhal` or if we update the wizard to get them.

My vote is to have them directly as dependencies.

### CODE REVIEW

File: packages/connector-edge/package.json [hunk 1]
```diff
@@ -11,12 +11,12 @@
     "eslint-plugin-typescript": "^0.8.1",
     "markdownlint-cli": "^0.7.0",
     "npm-run-all": "^4.1.2",
-    "sonarwhal": "^0.24.0",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/connector-edge/package.json]: > 0.26.0

File: packages/connector-edge/package.json [hunk 2]
```diff
@@ -11,12 +11,12 @@
...
     "typescript": "^2.6.2",
     "typescript-eslint-parser": "^13.0.0"
   },
   "peerDependencies": {
-    "sonarwhal": "^0.24.0"
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/connector-edge/package.json]: > 0.26.0

File: packages/formatter-codeframe/package.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-markdown": "^1.0.0-beta.7",
+    "eslint-plugin-typescript": "^0.8.1",
+    "markdownlint-cli": "^0.7.0",
+    "npm-link-check": "^2.0.0",
+    "npm-run-all": "^4.1.2",
+    "nyc": "^11.4.1",
+    "proxyquire": "^1.8.0",
+    "rimraf": "^2.6.2",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/formatter-codeframe/package.json]: > 0.26.0

File: packages/formatter-codeframe/package.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  ],
+  "license": "Apache-2.0",
+  "main": "dist/src/codeframe.js",
+  "name": "@sonarwhal/formatter-codeframe",
+  "nyc": {
+    "branches": 80,
+    "extends": "../../../../.nycrc"
+  },
+  "peerDependencies": {
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/formatter-codeframe/package.json]: > 0.26.0

File: packages/formatter-json/package.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-markdown": "^1.0.0-beta.7",
+    "eslint-plugin-typescript": "^0.8.1",
+    "markdownlint-cli": "^0.7.0",
+    "npm-link-check": "^2.0.0",
+    "npm-run-all": "^4.1.2",
+    "nyc": "^11.4.1",
+    "proxyquire": "^1.8.0",
+    "rimraf": "^2.6.2",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/formatter-json/package.json]: > 0.26.0

File: packages/formatter-json/package.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    "json"
+  ],
+  "license": "Apache-2.0",
+  "main": "dist/src/json.js",
+  "name": "@sonarwhal/formatter-json",
+  "nyc": {
+    "extends": "../../../../.nycrc"
+  },
+  "peerDependencies": {
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/formatter-json/package.json]: > 0.26.0

File: packages/formatter-stylish/package.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-markdown": "^1.0.0-beta.7",
+    "eslint-plugin-typescript": "^0.8.1",
+    "markdownlint-cli": "^0.7.0",
+    "npm-link-check": "^2.0.0",
+    "npm-run-all": "^4.1.2",
+    "nyc": "^11.4.1",
+    "proxyquire": "^1.8.0",
+    "rimraf": "^2.6.2",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/formatter-stylish/package.json]: > 0.26.0

File: packages/formatter-stylish/package.json [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    "stylish"
+  ],
+  "license": "Apache-2.0",
+  "main": "dist/src/stylish.js",
+  "name": "@sonarwhal/formatter-stylish",
+  "nyc": {
+    "extends": "../../../../.nycrc"
+  },
+  "peerDependencies": {
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/formatter-stylish/package.json]: > 0.26.0

File: packages/formatter-summary/package.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-markdown": "^1.0.0-beta.7",
+    "eslint-plugin-typescript": "^0.8.1",
+    "markdownlint-cli": "^0.7.0",
+    "npm-link-check": "^2.0.0",
+    "npm-run-all": "^4.1.2",
+    "nyc": "^11.4.1",
+    "proxyquire": "^1.8.0",
+    "rimraf": "^2.6.2",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/formatter-summary/package.json]: > 0.26.0

File: packages/formatter-summary/package.json [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    "summary"
+  ],
+  "license": "Apache-2.0",
+  "main": "dist/src/summary.js",
+  "name": "@sonarwhal/formatter-summary",
+  "nyc": {
+    "extends": "../../../../.nycrc"
+  },
+  "peerDependencies": {
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/formatter-summary/package.json]: > 0.26.0

File: packages/parser-javascript/package.json [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-markdown": "^1.0.0-beta.7",
+    "eslint-plugin-typescript": "^0.8.1",
+    "markdownlint-cli": "^0.7.0",
+    "npm-link-check": "^2.0.0",
+    "npm-run-all": "^4.1.2",
+    "nyc": "^11.4.1",
+    "proxyquire": "^1.8.0",
+    "rimraf": "^2.6.2",
+    "sonarwhal": "^0.25.0",
```
Comment (alrra) [inline packages/parser-javascript/package.json]: > 0.26.0

File: packages/parser-javascript/package.json [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    "javascript"
+  ],
+  "license": "Apache-2.0",
+  "main": "dist/src/javascript.js",
+  "name": "@sonarwhal/parser-javascript",
+  "nyc": {
+    "extends": "../../../../.nycrc"
+  },
+  "peerDependencies": {
+    "sonarwhal": "^0.25.0"
```
Comment (alrra) [inline packages/parser-javascript/package.json]: > 0.26.0

File: packages/formatter-json/README.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+# Formatter json (`@sonarwhal/formatter-json`)
```
Comment (alrra) [inline packages/formatter-json/README.md]: With rules we [only specified the name of the rule between `()`](https://github.com/sonarwhal/sonarwhal/tree/master/packages/rule-apple-touch-icons#require-an-apple-touch-icon-apple-touch-icons). We should be consistent and either use only the name here, or update things everywhere.

Comment (molant) [reply packages/formatter-json/README.md]: I think we should use the name of the package between `()`

Comment (alrra) [reply packages/formatter-json/README.md]: Can you also update the rules then? Thanks!

**Comment by molant** [comment]
This should be ready to merge or at least be reviewed. We might want to wait until #832 is done or at least until we publish the next version that checks if a rule is installed or not that @sarvaje is working on.

**Comment by sarvaje** [comment]
This is blocked because we need to publish first https://github.com/sonarwhal/sonarwhal/pull/834 when it is finish.

**Comment by molant** [comment]
Closing in favor of #844

**Pull request closed without merging**